File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        p1, p2 = 0, 0
        res = 0
        st = set()
        while p2 < len(s):
            while s[p2] in st:
                st.remove(s[p1])
                p1 += 1
            st.add(s[p2])
            res = max(res, p2 - p1 + 1)
            p2 += 1
        return res

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int p1 = 0;
        int p2 = heights.size() - 1;
        int ans = min(heights[p1], heights[p2]) * (heights.size() - 1);
        while(p1 < p2){
            if(heights[p1] < heights[p2]){
                p1++;
            } else{
                p2--;
            }
            ans = max(min(heights[p1], heights[p2]) * (p2-p1), ans);
        }
        return ans;
    }
};
